juju/juju
# undertaker watches namespace for caas

## Description of change

1. undertaker watches namespace for caas model;
2. fixed a few related bugs;

## QA steps

1. prepare a caas model;
2. deploy some workload;
3. juju destroy-model t1 --debug -y --destroy-storage

Now, we will see the destroy cmd waits until namespace's terminated.

## Documentation changes

No document change is required because it's internal logic change.
